### PR TITLE
TSK-1676: Fixed REST Exception Mapping for *InUseExceptions.

### PR DESCRIPTION
**Pull request by holgerhagen**
---
 
### For the submitter:
- [ ] I updated the [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview) and will supply links to the specific files
- [x] I did not update the [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview)
- [ ] I included a link to the [SonarCloud branch analysis](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1019969636/SonarCloud+Integration)
- [ ] I added a description of changes on the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana)
- [x] I did not update the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana)
- [x] I put the ticket in review
- [ ] After integration of the pull request, I verified our [bluemix test environment](http://taskana.mybluemix.net/taskana) is not broken

### Verified by the reviewer:
- [ ] Commit message format → TSK-XXX: Your commit message.
- [ ] Submitter's update to [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview) is sufficient
- [ ] SonarCloud analysis meets our standards
- [ ] Update of the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana) reflects changes
- [ ] PR fulfills the ticket
- [ ] Edge cases and unwanted side effects are tested
- [ ] Readability

### CODE REVIEW

File: rest/taskana-rest-spring/src/test/java/pro/taskana/classification/rest/ClassificationControllerIntTest.java [hunk 1]
```diff
@@ -425,7 +425,34 @@ void testDeleteClassification() {
...
+    assertThatThrownBy(httpCall)
+        .isInstanceOf(HttpStatusCodeException.class)
+        .extracting(HttpStatusCodeException.class::cast)
+        .extracting(HttpStatusCodeException::getStatusCode)
+        .isEqualTo(HttpStatus.NOT_FOUND);
+  }
+
+  @Test
+  @DirtiesContext
+  void should_ReturnStatusCodeLocked_WhenDeletingClassificationInUse() {
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring/src/test/java/pro/taskana/classification/rest/ClassificationControllerIntTest.java:437]:
```suggestion
  void should_ReturnStatusCodeLocked_When_DeletingClassificationInUse() {
```

File: rest/taskana-rest-spring/src/test/java/pro/taskana/classification/rest/ClassificationControllerIntTest.java [hunk 2]
```diff
@@ -425,7 +425,34 @@ void testDeleteClassification() {
...
+    String url =
+        restHelper.toUrl(
+            RestEndpoints.URL_CLASSIFICATIONS_ID, "CLI:000000000000000000000000000000000003");
+    HttpEntity<String> auth = new HttpEntity<>(RestHelper.generateHeadersForUser("businessadmin"));
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring/src/test/java/pro/taskana/classification/rest/ClassificationControllerIntTest.java:441]:
```suggestion
    HttpEntity<?> auth = new HttpEntity<>(RestHelper.generateHeadersForUser("businessadmin"));
```

File: rest/taskana-rest-spring/src/test/java/pro/taskana/classification/rest/ClassificationControllerIntTest.java [hunk 3]
```diff
@@ -425,7 +425,34 @@ void testDeleteClassification() {
...
+
+    ThrowingCallable httpCall =
+        () ->
+            TEMPLATE.exchange(
+                url,
+                HttpMethod.DELETE,
+                auth,
+                ParameterizedTypeReference.<ClassificationSummaryRepresentationModel>forType(
+                    ClassificationSummaryRepresentationModel.class));
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring/src/test/java/pro/taskana/classification/rest/ClassificationControllerIntTest.java:450]:
```suggestion
                ParameterizedTypeReference.forType(ClassificationSummaryRepresentationModel.class));
```

**Review by mustaphazorgati — CHANGES_REQUESTED** [review]

**Review by mustaphazorgati — APPROVED** [review]
